adal/src/main/java/com/microsoft/aad/adal/Discovery.java

```diff
@@ -212,6 +224,9 @@ private void performInstanceDiscovery(final URL authorityUrl, final String trust
             return;
         }
 
+        //Check if the network connection available
+        HttpWebRequest.throwIfNetworkNotAvailable(mContext);
```

Comment:
This could be `null`, couldn't it?

Reply:
We need to call the setter when creating the Discovery instance. This is the best solution I could found so far.

Reply:
Let's get rid of the setter.